app.py

```diff
@@ -15,45 +15,46 @@ def index():
     # If form is submitted
     if request.method == 'POST':
         NotamRequest = Models.NotamRequest(request.form)
-
-        # get lat/long of airports
-        NotamRequest.startLat, NotamRequest.startLong = alc.get_lat_and_lon(NotamRequest.startAirport)
-        NotamRequest.destLat, NotamRequest.destLong = alc.get_lat_and_lon(NotamRequest.destAirport)
+        airports = [NotamRequest.startAirport, NotamRequest.destAirport, NotamRequest.additionalAirport1, NotamRequest.additionalAirport2, NotamRequest.additionalAirport3 ]
+        apiOutputs = []
+        i= 0
         
-        NotamRequest.radius = int(NotamRequest.radius)
-        NotamRequest.pathWidth = int(NotamRequest.pathWidth)
-
-        # get the list of coordinates that need to be called to cover area
-        coordList = MinimalCirclesPath.getPath(NotamRequest.startLat, 
-                                               NotamRequest.startLong,
-                                               NotamRequest.destLat,
-                                               NotamRequest.destLong, 
-                                               NotamRequest.radius, # circle radius
-                                               NotamRequest.pathWidth) # path width
+        while i != 4 and airports[i+1] is not None:
+            
+            # get lat/long of airports
+            NotamRequest.startLat, NotamRequest.startLong = alc.get_lat_and_lon(airports[i])
+            NotamRequest.destLat, NotamRequest.destLong = alc.get_lat_and_lon(airports[i+1])
+        
+            NotamRequest.radius = int(NotamRequest.radius)
+            NotamRequest.pathWidth = int(NotamRequest.pathWidth)
 
-        # start timer
-        startTime = time.time() 
+            # get the list of coordinates that need to be called to cover area
+            coordList = MinimalCirclesPath.getPath(NotamRequest.startLat, 

```

Comment:
we already have NOTAMs for `airports[i]` if it was the destination of the previous leg of the flight, but now is the `startLat` and `startLong`, which means we're effectively making two API calls where only one would be needed

Reply:
I understand on this on this one, Sir. I'll have to figure out how to move the start point to 100 nm down the new path.

Reply:
my initial thought is that you could use the last point in coordlist as the first point for the next segment

Reply:
although if we are going to parallelize this process we would want to recalculate it in each loop so there is no cross loop dependencies. We could use calculateBearing() and nextPoint() in MinimalCircles.py to achieve that